### DISCUSSION TITLE
Cache across different `useContractReads`

### DISCUSSION THREAD

**Discussion by Drewsapple** [wevm/wagmi#2400] [Q&A]
If I have multiple `useContractReads` like:

```
useContractReads({
  contracts: [
    a,
    b,
    c,
})
```

and

```
useContractReads({
  contracts: [
    b,
    c,
    d,
})
```

can I use a combination of `scopeKey`, `cacheTime`, and `staleTime` so that the contract calls `b` and `c` are only fetched from the RPC once (per N seconds)?

**Comment by jxom** [comment]
You could probably just use `useContractRead` by itself with those config, as wagmi v1 batches up multiple `useContractRead`s in a 16ms interval in a single `eth_call` via multicall3. Let me know if that works!

**Reply by Drewsapple** [reply]
In my use case, I can't use a single read. The first set of reads is used to aggregate a list of erc20 balances across multiple holder addresses, to sum per token across all of them. The second set of reads is inside a component that shows the token balances for a single holder address.

I currently pass the balances as props to the second component, but it necessitates some ugly code that slices the first list from `index * tokenListLength` to `(index + 1) * tokenListLength` when passing the balances. Array indexing seems more fragile than each component asking for the data it needs, and makes reordering these components harder.
